daemon/cmd/daemon_main.go

```diff
@@ -1918,6 +1919,45 @@ func startDaemon(d *Daemon, restoredEndpoints *endpointRestoreState, cleaner *da
 	if err != nil {
 		log.WithError(err).Error("Unable to store Viper's configuration")
 	}
+
+	checkEndpointBPFProgs(d)
+}
+
+// checkEndpointBPFProgs check to ensure endpoints associated BPF programs remain loaded.
+func checkEndpointBPFProgs(d *Daemon) {
```

Comment:
This is taking us backwards with regards to modularizing the daemon. This should be implemented as a standalone cell that depends on `Promise[*Daemon]` (or a bridged interface) and `endpointmanager.EndpointsLookup`. It does make it harder/impossible to backport beyond v1.14 though, so not 100% sure whether we should take this in in this form and then refactor in follow-up after backports...

Reply:
For backports, we can come up with a separate implementation that's as close as possible to the version that lands in main, but I agree with this feedback of using the new mechanisms that enable modularization.

Reply:
I've reworked the original proposal. Let see if that alleviate some of your concerns...